examples/tests/ngtcp2test/client.py

```diff
@@ -82,11 +85,11 @@ def _assert_hs(self, c_hs, s_hs):
         assert self.server.hs_stripe == s_hs, \
             f'Expected "{s_hs}", got "{self.server.hs_stripe}"\n'
 
-    def assert_non_resume_handshake(self):
+    def assert_non_resume_handshake(self, allow_hello_retry=True):
```

Comment:
Is allow_hello_retry a new parameter?  It looks like it is not used anywhere and always True.

Reply:
I used it to test the recent improvements in wolfSSL. I think, given we configure clients/servers by default with the same ciphers/groups, the tests should verify in the future that the handshake succeeds without an additional roundtrip.

Reply:
So this is the feature that the future improvements use to disallow HRR.

Reply:
The HRR case that happend (fixed now in wolfssl master), was between boringssl server and wolfssl client. wolfssl sent a KEY_SHARE with a group that was not the highest in SUPPORTED_GROUPS. boringssl only looked for the highest group match, found no key share and triggered HRR.

Reply:
Yeah, this HRR scenario happens between implementations, and the thing is we do not save a group used in the initial connection.  We just workaround it to set the same group preference to all implementations.